src/sentry/api/endpoints/organization_release_details.py

```diff
@@ -64,10 +80,42 @@ def __get_next_release_date_query_q_and_order_by(release):
             "order_by": ["-date_added", "-id"],
         }
 
+    @staticmethod
+    def __filter_down_snuba_primary_results_according_to_status_and_query(
+        org,
+        project_ids,
+        version_list,
+        status_filter,
+    ):
+        """
+        Helper function used to query Release model from the primary results of a snuba query
+        which happens when sorting on sessions, users, crash_free_users and crash_free_sessions
+        Inputs:-
+            * org: organization
+            * project_ids: list of project ids
+            * version_list: list of release versions
+            * status_filter: either open or archived
+        Returns:-
+            A list of filtered release versions in the same order it received it
+        """
+        queryset = Release.objects.filter(
+            organization=org, projects__id__in=project_ids, version__in=version_list
+        )
+
+        # Add status filter
+        queryset = add_status_filter_to_queryset(queryset, status_filter)
+
+        # Required re ordering because django filter does not guarantee order of snuba primary order
+        release_list = list(queryset)
+        release_list.sort(key=lambda release: version_list.index(release.version))
```

Comment:
I guess the number of items here is small, but won't this be a fairly inefficient sort? `.index` is linear.

Reply:
So I updated this with I would say better performing logic (not sure how much better) but just to highlight something here, this release_list is going to have at most 20 elements -> https://github.com/getsentry/sentry/blob/master/src/sentry/snuba/sessions.py#L621 

so I think with the new change it would be fine